docs: Portal 1.17.1 release notes (#1784)
TykTechnologies/tyk-docs: developer-support/release-notes/portal.mdx

## 1.17 Release Notes

### 1.17.1 Release Notes 

#### Release Date 24 March 2026

#### Release Highlights

Tyk Developer Portal has been updated to Golang 1.25 and Debian 13 (Trixie) for enhanced security and performance. This patch also resolves some issues with the Portal API and synchronization of assets with the Tyk Dashboard.

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.17.1) below.

#### Breaking Changes

There are no breaking changes in this release.

#### Deprecations

There are no deprecations in this release.

#### Upgrade instructions
<a id="upgrade-1.17.1"></a>

If you are upgrading to 1.17.1, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker Image to pull](https://hub.docker.com/r/tykio/portal/tags?page=&page_size=&ordering=&name=v1.17.1)
  - ```bash
    docker pull tykio/portal:v1.17.1
    ```
- Helm charts
  - [tyk-charts](/developer-support/release-notes/helm-chart)

#### Changelog
<a id="Changelog-v1.17.1" data-scroll-offset></a>

##### Changed

<AccordionGroup>

> <Accordion title='Updated Developer Portal to Go 1.25'>

Reviewer: ```suggestion
<Accordion title='Update Developer Portal to Golang 1.25'>
```

> Updated the Tyk Developer Portal to use Go 1.25, ensuring compatibility with the latest Go runtime features and reducing exposure to potential security vulnerabilities in older versions.

Reviewer: ```suggestion
Updated the Tyk Developer Portal to use Golang 1.25, ensuring compatibility with the latest Golang runtime features and reducing exposure to potential security vulnerabilities in older versions.
```

The Portal now correctly detects and propagates all plan field changes to Dashboard Policies, ensuring that updates to key expiration settings and metadata are immediately reflected in the Dashboard regardless of whether other fields are modified at the same time.
</Accordion>

<Accordion title='Fixed policy reference inconsistency in API definitions'>
Resolved an issue where the Portal incorrectly used database object IDs (_id) instead of custom Policy IDs (id) when linking policies to APIs through Products or Plans. This caused "policy not found" errors during JWT validation as the Gateway prioritizes custom Policy IDs. The Portal now consistently uses Policy IDs for all scope-to-policy mappings and properly cleans up old mappings during updates.

</AccordionGroup>

##### Security Fixes

<AccordionGroup>

<Accordion title='CVE fixed'>
Addressed the following CVEs, providing increased protection against security
vulnerabilities, including, but not limited to:

> - <a href="https://nvd.nist.gov/vuln/detail/CVE-2026-33186" target="_blank">CVE-2026-33186</a

Reviewer: ```suggestion
- <a href="https://nvd.nist.gov/vuln/detail/CVE-2026-33186" target="_blank">CVE-2026-33186</a>
```

## 1.17 Release Notes

### 1.17.1 Release Notes 

#### Release Date 24 March 2026

#### Release Highlights

> Tyk Developer Portal fixes CVEs and has been updated to Go 1.25 and Debian 13 (Trixie) for enhanced security and performance. This release also resolves several Portal API issues, including incorrect `Content-Type` headers, improper null value handling for enum fields, and some policy reference inconsistencies.

Reviewer 2: ```suggestion
Tyk Developer Portal has been updated to Golang 1.25 and Debian 13 (Trixie) for enhanced security and performance. This patch also resolves some issues with the Portal API and synchronization of assets with the Tyk Dashboard.
```

For a comprehensive list of changes, please refer to the detailed [changelog](#Changelog-v1.17.1) below.

#### Breaking Changes

There are no breaking changes in this release.

> #### Dependencies

Reviewer 2: ```suggestion
```

> <a id="dependencies-1.17.1"></a>

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

> | Portal Version | Recommended Releases | Backwards Compatibility |

Reviewer 2: ```suggestion
```

> |--------|-------------------|-------------|

Reviewer 2: ```suggestion
```

> | 1.17.1 | MDCB vX.Y.Z       | MDCB vX.Y.Z |

Reviewer 2: ```suggestion
```

> |        | Operator vX.Y.Z   | Operator vX.Y |

Reviewer 2: ```suggestion
```

> |        | Sync vX.Y.Z       | Sync vX.Y.Z |

Reviewer 2: ```suggestion
```

> |        | Helm Chart vX.Y   | Helm all versions |

Reviewer 2: ```suggestion
```

> |        | EDP vX.Y.Z        | EDP all versions |

Reviewer 2: ```suggestion
```

> |        | Pump vX.Y.Z       | Pump all versions |

Reviewer 2: ```suggestion
```

> |        | TIB (if using standalone) vX.Y.Z | TIB all versions |

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

> ##### 3rd Party Dependencies & Tools

Reviewer 2: ```suggestion
```

> <a id="3rdPartyTools-v1.17.1"></a>

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

> | Third Party Dependency | Tested Versions | Compatible Versions | Comments |

Reviewer 2: ```suggestion
```

> | ---------------------- | --------------- | ------------------- | -------- |

Reviewer 2: ```suggestion
```

> | [GoLang](https://go.dev/dl/)          | X.YZ | X.YZ | [Go plugins](/api-management/plugins/golang) must be built using Go X.YZ |

Reviewer 2: ```suggestion
```

> | [Redis](https://redis.io/download/)   | 6.x, 7.x    |  6.x, 7.x    | |

Reviewer 2: ```suggestion
```

> | [Valkey](https://valkey.io/download/) | 8.0.x, 8.1.x    | 7.2.x, 8.0.x, 8.1.x    | |

Reviewer 2: ```suggestion
```

> | [MongoDB](https://www.mongodb.com/try/download/community)  | 6, 7, 8  | 5, 6, 7, 8  | |

Reviewer 2: ```suggestion
```

> | [DocumentDB](https://aws.amazon.com/documentdb/)  | 4, 5  | 4, 5  | |

Reviewer 2: ```suggestion
```

> | [PostgreSQL](https://www.postgresql.org/download/) | 13.x - 17.x | 13.x - 17.x  | |

Reviewer 2: ```suggestion
```

> | [OpenAPI Specification](https://spec.openapis.org/oas/v3.1.2.html) | v3.1.x | v3.1.x | Supported by [Tyk OAS](/api-management/gateway-config-tyk-oas)|

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

> Given the potential time difference between your upgrade and the release of this version, we recommend users verify the ongoing support of third-party dependencies they install, as their status may have changed since the release.

Reviewer 2: ```suggestion
```

> 

Reviewer 2: ```suggestion
```

#### Deprecations

There are no deprecations in this release.

#### Upgrade instructions
<a id="upgrade-1.17.1"></a>

If you are upgrading to 1.17.1, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker Image to pull](https://hub.docker.com/r/tykio/portal/tags?page=&page_size=&ordering=&name=v1.17.1)
  - ```bash
    docker pull tykio/portal:v1.17.1
    ```
- Helm charts
  - [tyk-charts](/developer-support/release-notes/helm-chart)

#### Changelog
<a id="Changelog-v1.17.1" data-scroll-offset></a>

##### Changed

<AccordionGroup>

<Accordion title='Updated Developer Portal to Go 1.25'>
Updated the Tyk Developer Portal to use Go 1.25, ensuring compatibility with the latest Go runtime features and reducing exposure to potential security vulnerabilities in older versions.
</Accordion>

<Accordion title='Update base images to Debian 13 (Trixie)'>
Updated the base Docker images for Tyk Portal to Debian 13 (Trixie) to address multiple vulnerabilities in the underlying operating system.
</Accordion>

</AccordionGroup>

##### Fixed

<AccordionGroup>

<Accordion title='Fixed Portal API Endpoints Returning Incorrect `Content-Type` Headers'>

> Resolved an issue where multiple Portal API endpoints returned incorrect `Content-Type` headers. Previously, endpoints that didn't explicitly write a response body would default to `text/plain` instead of the expected `application/json`, potentially causing client applications to misinterpret API responses.

Reviewer 2: ```suggestion
Resolved an issue where several Portal API endpoints returned incorrect `Content-Type` headers. Previously, endpoints that didn't explicitly write a response body would default to `text/plain` instead of the expected `application/json`, potentially causing client applications to misinterpret API responses.
```

The fix addresses this issue across all affected Portal API endpoints, including product management endpoints (documentation, logos, previews, OpenAPI specs, GraphQL schemas, tags), user management endpoints, and other Portal API operations. All Portal API endpoints now return the correct `Content-Type` header that matches their actual response content.
</Accordion>

<Accordion title='Fixed Portal API Enum Field Handling for Null Values'>

> Resolved an issue where the Portal API returned empty strings (`""`) instead of `null` values for optional enum fields when their values were not set. Previously, endpoints like `/api/posts` and `/api/catalogues` would incorrectly serialize null enum values as empty strings, violating the API contract and potentially causing client-side parsing issues.

Reviewer 2: ```suggestion
Resolved an issue where the Portal API returned empty strings (`""`) instead of `null` values for optional enum fields when their values were not set. Previously, endpoints such as `/api/posts` and `/api/catalogues` would incorrectly serialize null enum values as empty strings, violating the API contract and potentially causing client-side parsing issues.
```

The Portal API now properly omits optional enum fields from responses when they are empty, ensuring consistent and correct API behavior across all endpoints, including posts, catalogues, access requests, and providers.
</Accordion>

<Accordion title='Fixed Plan Field Updates Not Propagating to Dashboard Policies'>

> Resolved an issue where updates to certain Plan fields in the Developer Portal were not properly propagating to their corresponding Dashboard Policies. Previously, when administrators updated only the `KeyExpiresIn` (key expiration time) or `MetaData` fields through either the Portal UI or API, these changes would not sync to the Dashboard unless other fields like rate limits or quotas were modified simultaneously. This occurred due to an incomplete change detection mechanism that omitted these fields from its validation checks. 

Reviewer 2: ```suggestion
Resolved an issue where updates to certain Plan fields in the Developer Portal were not properly propagating to their corresponding Dashboard Policies. Previously, when administrators updated only the `KeyExpiresIn` (key expiration time) or `MetaData` fields through either the Portal UI or API, these changes would not sync to the Dashboard unless other fields such as rate limits or quotas were modified simultaneously. This occurred due to an incomplete change detection mechanism that omitted these fields from its validation checks. 
```

The Portal now correctly detects and propagates all plan field changes to Dashboard Policies, ensuring that updates to key expiration settings and metadata are immediately reflected in the Dashboard regardless of whether other fields are modified at the same time.
</Accordion>

<Accordion title='Fixed policy reference inconsistency in API definitions'>
Resolved an issue where the Portal incorrectly used database object IDs (_id) instead of custom Policy IDs (id) when linking policies to APIs through Products or Plans. This caused "policy not found" errors during JWT validation as the Gateway prioritizes custom Policy IDs. The Portal now consistently uses Policy IDs for all scope-to-policy mappings and properly cleans up old mappings during updates.

> </AccordionGroup>

Reviewer: Missing `</Accordion>` closing tag for the "Fixed policy reference inconsistency in API definitions" accordion, this will break MDX rendering.

```suggestion
</Accordion>

</AccordionGroup>
```
